Common/Model.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Mathematics;
+using Assimp;
+using System.Drawing;
+using System.Drawing.Imaging;
+using PixelFormat = OpenTK.Graphics.OpenGL4.PixelFormat;
+using TextureWrapMode = OpenTK.Graphics.OpenGL4.TextureWrapMode;
+using AssimpMesh = Assimp.Mesh;
+
+namespace LearnOpenTK.Common
+{
+    public class Model
+    {
+        // model data
+        private List<Mesh> meshes;
+
+        // constructor, expects a filepath to a 3D model.
+        public Model(string path)
+        {
+            LoadModel(path);
+        }
+
+        // draws the model, and thus all its meshes
+        public void Draw()
+        {
+            for (int i = 0; i < meshes.Count; i++)
+                meshes[i].Draw();
+        }
+
+        // loads a model with supported ASSIMP extensions from file and stores the resulting meshes in the meshes list.
+
+        private void LoadModel(string path)
+        {
+            //Create a new importer
+            AssimpContext importer = new AssimpContext();
+
+            //This is how we add a logging callback 
+            LogStream logstream = new LogStream(delegate (String msg, String userData)
+            {
+                Console.WriteLine(msg);
+            });
+            logstream.Attach();
+
+            //Import the model. All configs are set. The model
+            //is imported, loaded into managed memory. Then the unmanaged memory is released, and everything is reset.
+            Scene scene = importer.ImportFile(path, PostProcessSteps.Triangulate);
+
+            // check for errors
+            if (scene == null || scene.SceneFlags.HasFlag(SceneFlags.Incomplete) || scene.RootNode == null)
+            {
+                Console.WriteLine("Unable to load model from: " + path);
+                return;
+            }
+            
+            //Reset the meshes and textures
+            meshes = new List<Mesh>();
+
+            //Set the scale of the model
+            float scale = 1/200.0f;
+            Matrix4x4 scalingMatrix = new Matrix4x4(scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, 1);
+
+            // process ASSIMP's root node recursively. We pass in the scaling matrix as the first transform
+            ProcessNode(scene.RootNode, scene, scalingMatrix);
+
+            importer.Dispose();
```

Comment:
Comment on this?

Reply:
I added in some comments, although I am not sure how necessary this is since I would think that the garbage collector would get this. This comes directly from the [Assimp-Net Wiki](https://code.google.com/archive/p/assimp-net/wikis/GettingStarted.wiki). Maybe it is because it was built off of the C++ version where you would need to manage your memory?

Reply:
Yeah possibly. I'm happy with the comment you added 🙂